src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -34,17 +40,61 @@ export class ResultsViewPageStore {
 
     @observable ajaxErrors: Error[] = [];
 
-    @observable studyId: string = '';
-    @observable sampleListId: string|null = null;
+    //@observable studyId: string = '';
+    //@observable sampleListId: string|null = null;
     @observable hugoGeneSymbols: string[]|null = null;
-    @observable sampleList: string[]|null = null;
+    //@observable sampleList: string[]|null = null;
+    @observable _studyToSampleIds:{[studyId:string]:{[sampleId:string]:boolean}}|null = null;
+    @observable studyToSampleListId:{[studyId:string]:string}|null = null;
+
+    readonly studyToSampleIds = remoteData<{[studyId:string]:{[sampleId:string]:boolean}}>(async()=>{
+        if (this._studyToSampleIds) {
```

Comment:
yeah, i would love to have some computed mode property that would hide this.  like, mode:'customSampleList'|'definedSampleList'
(maybe there's something better than 'definedSampleList')
My brain has a hard time distinguishing sampleId from sampleListId

Reply:
I think that may not be future proof - eg one study could have a sample list id, another a custom sample list